### ISSUE TITLE
Ensure lower-case secret names when auto naming

### ISSUE THREAD

**Issue by roderik** [pulumi/pulumi-kubernetes#1652]
## Steps to reproduce

Deploying the following secret

```
        const pullSecret = new k8s.core.v1.Secret(
          `${baseName}-pullSecret`,
          {
            type: 'kubernetes.io/dockerconfigjson',
            metadata: {
              namespace: namespace.metadata.name,
            },
            stringData: {
              '.dockerconfigjson': pulumi
                .all([config.require<string>('githubUsername'), config.requireSecret<string>('ghcrPullSecretPassword')])
                .apply(([username, password]) =>
                  JSON.stringify({
                    auths: {
                      'ghcr.io': {
                        auth: Buffer.from(`${username}:${password}`).toString('base64'),
                      },
                    },
                  })
                ),
            },
          },
          {
            provider,
          }
        );
```

Failed with 

    error: resource mainnet/eks-europe-pullSecret-783hbmy3 was not successfully created by the Kubernetes API server : Secret "eks-europe-pullSecret-783hbmy3" is invalid: metadata.name: Invalid value: "eks-europe-pullSecret-783hbmy3": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')

I would assume that the auto name would be lowercased by default

**Comment by lukehoban** [comment]
> I would assume that the auto name would be lowercased by default

Yes - it should!

**Comment by mikhailshilkov** [comment]
> Yes - it should!

Should it always be lowercased? Does that apply to other providers too?

**Comment by lukehoban** [comment]
> Should it always be lowercased? Does that apply to other providers too?

Autonaming for a specific resource should always generate a legal name for that resource. We have many overrides to ensure this in the classic AWS and azure providers.

**Comment by blampe** [comment]
Fixed since at least #2137.
